client/my-sites/themes/thanks-modal.jsx

```diff
@@ -201,20 +201,22 @@ const ThanksModal = React.createClass( {
 
 export default connect(
 	( state, { site } ) => {
+		let siteIdOrWpcom;
 		if ( ! site ) {
-			return {};
+			siteIdOrWpcom = 'wpcom';
+		} else {
+			siteIdOrWpcom = isJetpackSite( state, site.ID ) ? site.ID : 'wpcom';
 		}
 
-		const siteIdOrWpcom = isJetpackSite( state, site.ID ) ? site.ID : 'wpcom';
-		const currentThemeId = getActiveTheme( state, site.ID );
+		const currentThemeId = getActiveTheme( state, site && site.ID );
```

Comment:
I think I'd rather put the guards in front of the selector, i.e.

```js
const currentThemeId = site && getActiveTheme( state, site.ID );
```

since we can't be entirely sure all these selectors won't break if they're given `false` as `siteId` arg. (Same below.)